src/index.ts

```diff
@@ -20,27 +22,19 @@ const RETRY_STATUSES = [ 408, 425, 429, 500,  501,  502,  503,  504,  505,  506,
 
 const canRetry = (status: number) => RETRY_STATUSES.includes(status)
 
-const ok = (status: number) => 200 <= status && status < 300
-
-const fetchWithRetry = (url: string, init: RequestInit, maxRetries: number = 3): Promise<void> => {
+const fetchWithRetry = (url: string, init: RequestInit, maxRetries: number = 3) => {
   let status = 500
-  const callFetch = (attempt: number = 0): Promise<void> =>
-    fetch(url, init).then((response) => {
+  const callFetch = (attempt: number = 0): Promise<any>=>
+    fetch(url, init).then((response: any) => {
       status = response.status
-      return ok(status)
-        ? { response, error: null }
-        : response.json()
-          .then((error) => ({ response, error }))
-          .catch(() => ({ response, error: { message: 'Unable to parse JSON' } }))
-    }).then(({ error }: any) => {
-      if (error) {
-        throw new Error(error.message || 'Unknown error')
-      }
+      return response.json()
```

Comment:
why removing the `ok` status ? What if session returns a 500 with a valid json ?